packages/js/core/src/types/Uri.ts

```diff
@@ -194,4 +209,26 @@ export class Uri {
   public toJSON(): string /* $ */ {
     return this._config.uri;
   }
+
+  private static inferAuthority(path: string): UriConfig | undefined {
+    let authority: string | undefined;
+
+    if (path.startsWith("https://")) {
```

Comment:
I think this should be generic. Effectively
```typescript
// Ignore this case, no inference needed
if (uri.startsWith("wrap://")) return;

// If the uri string starts with <schema>://
// infer that the authority == schema
const match = uri.match(/$[a-zA-Z0-9]+:\/\//);

if (match.length > 0) {
  authority = match[0];
}
```

This way we can turn `any://foo-bar-baz` into `wrap://any/foo-bar-baz`. NOTE: the `<schema>://` doesn't carry over, this is desired in all cases IMO.

Reply:
Warning: the regex above was not tested, and off the top of my head

Reply:
I will make it generalizable, sure thing. The HTTP resolver will need a simple update to support this. Currently, the HTTP resolver expects a full URL in the path. We just need to make sure it appends `https://` or `http://` in cases where it is missing.

Reply:
I updated the http uri resolver extension wrapper and sent a transaction to the Safe queue https://app.safe.global/transactions/queue?safe=eth:0xC327fEAd871d67476d27EFA941335A8b69e04e3A

Reply:
heads up: I accidently pushed the http resolver change to the main branch of the uri resolver extensions repo